modules/core/src/webgl/classes/texture-cube.js

```diff
@@ -46,16 +50,18 @@ export default class TextureCube extends Texture {
     //   }
     // });
 
-    this.setCubeMapImageData(props);
-
-    // Called here so that GL.
-    // TODO - should genMipmap() be called on the cubemap or on the faces?
-    if (mipmaps) {
-      this.generateMipmap(props);
-    }
-
     // Store props for accessors
     this.opts = props;
+
+    this.setCubeMapImageData(props).then(() => {
+      this.loaded = true;
+
+      // TODO - should genMipmap() be called on the cubemap or on the faces?
```

Comment:
Great if we can research this before we land?

Reply:
`genMipamap` should be called once on the cubemap. It's a per-texture operation.

Reply:
@tsherif can you explain?

Reply:
Why does this code work anyway?